### PR TITLE
Bug 1915661: update image-pruner role to include jobs, cronjobs and statefulsets

### PR DESCRIPTION
**Pull request by soltysh**
The oc code which added jobs, cronjobs and statefulset is in https://github.com/openshift/oc/pull/671
/assign @sttts

### CODE REVIEW

**Comment by soltysh** [comment]
@ricardomaraschini fyi, not sure why our tests didn't catch that sooner, it'd be reasonable to add a test if we don't have one.

**Comment by ricardomaraschini** [comment]
> @ricardomaraschini fyi, not sure why our tests didn't catch that sooner, it'd be reasonable to add a test if we don't have one.

Yes, I will work in getting a test in place for this (or check why our current tests did not catch it)

**Comment by ricardomaraschini** [comment]
@soltysh seems like we now have two PRs for the same thing https://github.com/openshift/openshift-apiserver/pull/175

File: test/testdata/bootstrappolicy/bootstrap_cluster_roles.yaml [hunk 1]
```diff
@@ -1542,6 +1556,14 @@ items:
     verbs:
     - get
     - list
+  - apiGroups:
+    - apps
+    - extensions
```
Comment (sttts) [inline test/testdata/bootstrappolicy/bootstrap_cluster_roles.yaml:1561]: extensions is still a thing?

Comment (soltysh) [reply test/testdata/bootstrappolicy/bootstrap_cluster_roles.yaml:1561]: It should not, there's only a handful of network resources still left there. Do you want to clean these permissions then?

Comment (sttts) [reply test/testdata/bootstrappolicy/bootstrap_cluster_roles.yaml:1561]: yes

**Comment by dmage** [comment]
/lgtm
Can we land it?

**Comment by soltysh** [comment]
> Can we land it?

@dmage I'm just updating the policies that @sttts requested and it should be read for merge later today

**Comment by soltysh** [comment]
/override ci/prow/e2e-cmd 
the failures are not related to this PR and worked on elsewhere.

**Pull request merged**